src/main/java/com/cegeka/academy/repository/UserChallengeRepository.java

```diff
@@ -0,0 +1,19 @@
+package com.cegeka.academy.repository;
+
+import com.cegeka.academy.domain.UserChallenge;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface UserChallengeRepository extends JpaRepository<UserChallenge, Long> {
+
+    Optional<UserChallenge> findById(Long id);
```

Comment:
Exista deja o metoda care iti aduce entitatea dupa id. Daca vrei sa iti returneze un Optional e in regula, daca nu vezi link.
https://docs.spring.io/spring-data/jpa/docs/current/api/org/springframework/data/jpa/repository/JpaRepository.html#getOne-ID-